android/src/toga_android/dialogs.py

```diff
@@ -22,7 +22,7 @@ def show(self, host_window, future):
 
         if self.native:
             # Show the dialog. Don't differentiate between app and window modal dialogs.
-            self.native.show()
+            self.native_alert_dialog = self.native.show()
```

Comment:
What is the purpose of this assignment? Firstly, `self.native` should be a reference to the native object; secondly [`Dialog.show()` returns void](https://developer.android.com/reference/android/app/Dialog#show()), so this value should always be None`.

Reply:
Actually `self.native` is a `AlertDialog.Builder` not a `AlertDialog`
https://github.com/beeware/toga/blob/3376c069cca85fb3c05523d3134f6842ed13f650/android/src/toga_android/dialogs.py#L32-L46
```
self.native = AlertDialog.Builder(toga.App.app.current_window._impl.app.native)
```
returns an `AlertDialog.Builder`: https://developer.android.com/reference/android/app/AlertDialog.Builder#Builder(android.content.Context)

So, doing `AlertDialog.Builder.show()`:  https://developer.android.com/reference/android/app/AlertDialog.Builder#show()
```
self.native_alert_dialog = self.native.show()
```
returns an `AlertDialog`


And `AlertDialog` has the `isShowing()` method: https://developer.android.com/reference/android/app/Dialog#isShowing()
which is used in the probe.